src/NLU.DevOps.CommandLine/EnumerableExtensions.cs

```diff
@@ -5,11 +5,40 @@ namespace NLU.DevOps.CommandLine
 {
     using System;
     using System.Collections.Generic;
+    using System.Diagnostics;
     using System.Linq;
     using System.Threading.Tasks;
 
     internal static class EnumerableExtensions
     {
+        public static Task<IEnumerable<TResult>> SelectAsync<T, TResult>(this IEnumerable<T> items, Func<T, Task<TResult>> selector, int degreeOfParallelism, int queriesPerSecond)
+        {
+            var stopwatch = Stopwatch.StartNew();
```

Comment:
Change to separate method that ramps up threads to max parallelism based on async throughput. E.g.,
```
1. Start timer loop that fires at rate limit (e.g., 2 requests per second fires ever 0.5 seconds)
2. For each timer fire, check how many active threads are running
2a. If max parallel threads are running, wait for some tasks to finish
2b. Otherwise start a new task
```